c/driver/common/CMakeLists.txt

```diff
@@ -21,6 +21,22 @@ set_target_properties(adbc_driver_common PROPERTIES POSITION_INDEPENDENT_CODE ON
 target_include_directories(adbc_driver_common PRIVATE "${REPOSITORY_ROOT}/c/include")
 target_link_libraries(adbc_driver_common PUBLIC nanoarrow::nanoarrow)
 
+# For static builds, we need to install the static library here so downstream
+# applications can link to it
+if(ADBC_BUILD_STATIC)
+  if(ADBC_WITH_VENDORED_NANOARROW)
+    message(WARNING "adbc_driver_common is not installed when ADBC_WITH_VENDORED_NANOARROW is ON. To use the static libraries, for now you must provide nanoarrow instead of using the vendored copy"
```

Comment:
What is the problem with installing in this case?

Reply:
Yeah, I tried to install Nanoarrow but CMake failed because of something about the header not being installable - decided to punt since to start with I don't think we need to support every possible case.